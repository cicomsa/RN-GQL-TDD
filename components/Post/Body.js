import React from 'react'
import { StyleSheet, Text, View } from 'react-native'

const Body = ({ body }) => {
  return (
    <View className="body" style={styles.post}>
      <Text className="body__copy" style={styles.body}>{body}</Text>
    </View>
  )
}

const styles = StyleSheet.create({
  post: {
    marginVertical: 5,
  },
  body: {
    color: 'orange',
    textAlign: 'center'
  }
})

export default Body