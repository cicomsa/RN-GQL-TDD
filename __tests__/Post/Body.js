import React from 'react'
import renderer from 'react-test-renderer'
import { mount, shallow } from 'enzyme'
import Body from '../../components/Post/Body.js'

const props = {
  body: 'body'
}

describe('Post - Author', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Body {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should return component', () => {
    const wrapper = mount(<Body {...props} />)
    expect(wrapper.find('.body').exists()).toBe(true)
  })
  it('should return correct copy', () => {
    const wrapper = shallow((<Body {...props} />)).dive()
    expect(wrapper.find('.body__copy').childAt(0).text()).toEqual(props.body)
  })
})
