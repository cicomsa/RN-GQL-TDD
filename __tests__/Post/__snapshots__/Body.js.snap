// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post - Body should render correctly 1`] = `
<View
  style={
    Object {
      "marginVertical": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "orange",
        "textAlign": "center",
      }
    }
  >
    
  </Text>
</View>
`;
